graylog2-server/src/main/java/org/graylog2/migrations/V20161125142400_EmailAlarmCallbackMigration.java

```diff
@@ -117,6 +117,11 @@ public void upgrade() {
     }
 
     private Optional<String> migrateStream(Stream stream) {
+        if (localAdminUser == null) {
+            throw new IllegalStateException("The root user is disabled. This migration will run only with the root " +
```

Comment:
This will throw an exception on every startup of the server. I think we can default to `"admin"` if the root user is disabled. We only pass the user ID (the name) to the alarm callback config and that alarm callback code is deprecated and will go away in the future.

Reply:
If I didn't get it wrong, it wouldn't throw on every startup but only if:
- the migration has never completed successfully before
- **and** there actually are streams which need migration
- **and** the root user is disabled

I wouldn't expect many installations getting into this situation so I thought it was tolerable to throw an exception in that case.
But looking into it again following your suggestion, the ID of the local admin user is fixed to "local:admin" regardless of the username configured in the config. So as the alarm callback code is going away anyways I'm now simply using this fixed string for the migration. Regardless of the activation/deactivation state of the root user.

Reply:
Thanks! :+1: